GiEnJul/ClientApp/src/components/admin/suggestedConnections/ConnectionSuggesterMacro.tsx

```diff
@@ -1,21 +1,153 @@
+  givers: [],
+  recipients: [],
+  selectedRecipient: initialSelection,
+  selectedGiver: initialSelection,
+  refreshing: false,
+};
+
+const ConnectionSuggesterMacro: React.FC<ConnectionSuggesterMacro> = ({
+  location,
+}) => {
+  const [state, setState] = useState(initialState);
+
+  const getSuggestedRecipients = () => {
+    setState({ ...state, refreshing: true });
+    api
+      .get("admin/Suggestions/Recipient", {
+        params: { location: location },
+      })
+      .then((response) => {
+        if (response.status == 200) {
+          setState({
+            ...state,
+            recipients: response.data,
+            refreshing: false,
+            selectedGiver: initialSelection,
+            selectedRecipient: initialSelection,
+          });
+        } else {
+          alert("Kunne ikke hente familier, prøv igjen");
+        }
+      });
+  };
+
+  useEffect(() => {
+    getSuggestedRecipients();
+  }, []);
+
+  //Fetch suggested givers, if recipients are updated
+  useEffect(() => {
+    api
+      .get("admin/Suggestions/Giver", {
+        params: { location: location },
+      })
+      .then((response) => {
+        if (response.status == 200) {
+          setState({ ...state, givers: response.data });
+        } else {
+          alert("Kunne ikke hente givere, prøv igjen");
+        }
+      });
+  }, [state.recipients]);
 
   const updateSelectedRecipient = (newSelected: [string, string]): void => {
-    setSelectedRecipient(newSelected);
+    setState({ ...state, selectedRecipient: newSelected });
+  };
+
+  const updateSelectedGiver = (newSelected: [string, string]): void => {
+    setState({ ...state, selectedGiver: newSelected });
+  };
+
+  const submitConnection = () => {
+    api
+      .post(
+        "admin/",
+        JSON.stringify({
+          GiverRowKey: state.selectedGiver[0],
+          GiverPartitionKey: state.selectedGiver[1],
+          RecipientRowKey: state.selectedRecipient[0],
+          RecipientPartitionKey: state.selectedRecipient[1],
+        })
+      )
+      .then((response) => {
+        if (response.status == 200) {
+          getSuggestedRecipients();
+        } else {
+          alert("Kunne ikke sende kobling, prøv igjen");
+        }
+      });
   };
 
   return (
     <>
-      <RecipientSuggestions selectRecipient={updateSelectedRecipient} />
+      <Grid container direction="row" justifyContent="center">
+        <RecipientSuggestions
+          selectRecipient={updateSelectedRecipient}
+          recipients={state.recipients}
+        />
+        <Container className="col-2">
+          <Button
+            fullWidth
+            variant="contained"
+            color="primary"
+            disabled={
+              state.selectedGiver === initialSelection ||
+              state.selectedRecipient === initialSelection
+            }
+            onClick={submitConnection}
+          >
+            Koble sammen
+          </Button>
+          <p></p>
```

Comment:
Is this \<p> tag necessary?

Reply:
Added styling on button and removed p tag